### DISCUSSION TITLE
Are there any plans to create a GUI like Unity or Godot?

### DISCUSSION THREAD

**Discussion by ohsalmeron** [bevyengine/bevy#11286] [Ideas]
Hi everyone!

As Bevy continues to grow into a powerful game engine, the desire for a user-friendly GUI becomes increasingly relevant. This proposal aims to address this need by introducing a flexible and intuitive visual editor, empowering developers, especially those visually oriented, to craft stunning experiences with greater ease.

Motivation:
While Bevy's ECS architecture offers tremendous power and control, it can present a steeper learning curve for beginners and designers. A robust GUI would bridge this gap by:

 Lowering the barrier to entry: Visual tools allow for intuitive exploration and manipulation, making game development more accessible to a wider audience.
 Boosting development speed: Drag-and-drop functionality and visual feedback can significantly expedite UI and scene construction.
 Empowering visual designers: Separating visual design from code allows designers to contribute without delving into complex code constructs.
 Enhancing the Bevy ecosystem: A powerful GUI complements the existing code-based workflow, providing a more holistic development experience.

Proposed Features:
The initial focus would be on core functionalities, gradually expanding based on community feedback. These include:

    Project File Explorer: A comprehensive file browser for intuitive asset and scene management.
    Scene Editor: A visual viewport for manipulating entities, components, and properties within the scene.
    Object Inspector: A dedicated panel for inspecting and editing individual object properties.
    Object Hierarchy: A tree-like view for managing object relationships and nesting.
    Basic UI Primitives: Implementation of commonly used UI elements like buttons, scroll views, and text fields.
    3D Primitives: Integration of basic 3D shapes for rapid scene prototyping.

Development Plan:
The project aims to follow an iterative and community-driven approach:

    Initial Proof-of-Concept: Develop a functional prototype showcasing core functionalities.
    Community Feedback: Gather feedback and suggestions from the Bevy community via discussions and surveys.
    Phased Implementation: Prioritize features based on community needs and technical feasibility.
    Open-source Development: Ensure full code transparency and encourage community contributions.

Next Steps:

This proposal seeks the community's interest and potential collaboration. To move forward, we need:
    Feedback: Share your thoughts on the proposed features and priorities.
    Contributors: Join the development effort by sharing expertise, technical skills, or design assets.
    Discussion: Let's spark a constructive conversation about the future of Bevy's visual development toolset.

**Comment by jdm** [comment]
See https://github.com/bevyengine/bevy/discussions/7100

**Reply by jdm** [reply]
Also https://github.com/bevyengine/bevy_editor_prototypes/discussions/1
